pyccel/codegen/printing/fcode.py

```diff
@@ -205,6 +206,7 @@ def __init__(self, parser, prefix_module = None):
 
         self._additional_code = None
         self._additional_imports = set([])
+        self._additional_declare = []
```

Comment:
Why do we need this? There are already ways to declare additional variables in this file. Search for `self._additional_code ` to see how this is handled elsewhere

Reply:
This is necessary to keep the declarations of the variables needed to loop over the matrix in the prelude. If they are added to the `self._additional_code`, they will raise the following error:
```
Error: Unexpected data declaration statement at (1)
.../__pyccel__/test.f90:16:26:

   16 |     integer(i64) :: i_0001
```

Reply:
This attribute is also used in ccode. I copied it from there.

Reply:
Note that I did not say to place the declaration in `self._additional_code` but simply to search for the variable in the existing code. If you do this you will see that there are other places in the code where new variables are declared but that their declaration does not require the use of `self._additional_code` or `self._additional_declare`. I am not sure why declaration of variables is handled differently in the 2 printers. I suspect it is related to scoping